rust/src/utils/distance/cosine.rs

```diff
@@ -101,41 +94,41 @@ fn cosine_dist_simd(from: &Float32Array, to: &Float32Array, dimension: usize) ->
     for y in to_values.chunks_exact(dimension) {
         #[cfg(any(target_arch = "aarch64"))]
         {
-            builder.append_value(unsafe { cosine_dist_neon(x, y, x_norm) });
+            builder.append_value(unsafe { 1.0 - cosine_dist_neon(x, y, x_norm) });
```

Comment:
This should happen inside the distance function OR the fn should be renamed cosine_similarity_neon/fma

Reply:
fixed.